### PR TITLE
[13.0][MIG] hr_expense_invoice

### PR DESCRIPTION
**Pull request by kongrattapong**
Standard migration, ready for review.
Supersede #702

### CODE REVIEW

**Comment by kongrattapong** [comment]
Q: How can I pass this context to invoice_line_ids
```
   'default_invoice_line_ids': [{'product_id': product_id,
                                 'name': name,
                                 'price_unit': unit_amount,
                                 'quantity': quantity, 'date_invoice': date,
                                 'account_id': account_id, 'invoice_line_tax_ids': tax_ids, }], }" />
                                 'quantity': quantity,
                                 'account_id': account_id,}]
```

File: hr_expense_invoice/models/account_move_line.py [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
     _inherit = "account.move.line"
 
-    @api.multi
     def reconcile(self, writeoff_acc_id=False, writeoff_journal_id=False):
         if self._context.get("use_hr_expense_invoice"):
             self = self.filtered(lambda l: not l.reconciled)
         res = super().reconcile(
             writeoff_acc_id=writeoff_acc_id, writeoff_journal_id=writeoff_journal_id
         )
+        for rec in self:
```
Comment (kittiu) [inline hr_expense_invoice/models/account_move_line.py]: ??? I am not sure we should do this. We should know more how system works

File: hr_expense_invoice/models/hr_expense.py [hunk 2]
```diff
@@ -40,5 +37,7 @@ def _get_account_move_line_values(self):
                     move_line[
                         "partner_id"
                     ] = expense.invoice_id.partner_id.commercial_partner_id.id
-                    move_line["account_id"] = expense.invoice_id.account_id.id
+                    move_line["account_id"] = expense.invoice_id.line_ids.filtered(
+                        "credit"
```
Comment (kittiu) [inline hr_expense_invoice/models/hr_expense.py]: Find account by using "credit" is not recommended. Better that you understand how Invoice get the account to use in move line, then use it.

**Comment by kittiu** [comment]
cc @Saran440

**Comment by kittiu** [comment]
> Q: How can I pass this context to invoice_line_ids
> 
> ```
>    'default_invoice_line_ids': [{'product_id': product_id,
>                                  'name': name,
>                                  'price_unit': unit_amount,
>                                  'quantity': quantity, 'date_invoice': date,
>                                  'account_id': account_id, 'invoice_line_tax_ids': tax_ids, }], }" />
>                                  'quantity': quantity,
>                                  'account_id': account_id,}]
> ```

On v12, it works like this. Still not sure if this is bug in v13. In the case it can't be resolved, I suggest you hide the option to create/edit from dropdown (user should not do that anyway).  `options="{'no_create':True}`

https://github.com/ecosoft-odoo/purchase-workflow/blob/12.0-add-purchase_work_acceptance/purchase_work_acceptance/models/purchase.py#L44-L52

**Comment by kittiu** [comment]
@Saran440 thanks for your final revision. I see some changes in https://github.com/OCA/hr/pull/725
So, would you help,

1. Help review/approve that this pull request -> https://github.com/OCA/hr/pull/725/files
2. Please redo the migration based of that PR (once it is merged)
3. Migrate hr_expense_cancel to v13

Note: I think this PR is quite messy, please close this PR and reopen a new one.

**Comment by Saran440** [comment]
change to this PR -> https://github.com/OCA/hr/pull/743

**Comment by pedrobaeza** [comment]
Can you please move this to OCA/hr-expense following #732?

**Comment by kongrattapong** [comment]
change to this PR -> #743

**Pull request closed without merging**